client/cli/src/params.rs

```diff
@@ -238,6 +238,21 @@ pub struct NetworkConfigurationParams {
 	#[allow(missing_docs)]
 	#[structopt(flatten)]
 	pub node_key_params: NodeKeyParams,
+
+	/// Enable flow control for yamux streams.
+	///
+	/// By default this is not enabled. When enabled, yamux ensures that senders
+	/// do not overwhelm receivers. Depending on the protocol, there is a risk
```

Comment:
Where do we write to a socket in a synchronous matter without reading? That doesn't sound like a proper async implementation.

Reply:
We don't. I'm not sure whether it's a good idea to actually tell that to users.
Rather than go into details, maybe the documentation should just say "This is an experimental feature that will be turned on by default once it's been tested.".

Reply:
Sounds like a good idea, could you change this @twittner?

Reply:
I can certainly replace the comment with an opaque wording if that is what you want. However I do not understand the reference to "synchronous write" here. The case can simply occur if both ends write concurrently more than their credit allows for and when done in  async/await style (e.g. `stream.write_all(...).await`)  there is no manual polling where they would do an interleaved read operation after a `Poll::Pending`.

Reply:
`stream.write_all(...).await` the same behavior would appear for any TCP socket.

Reply:
So, nothing special for yamux here.

Reply:
But what do you mean by "doesn't sound like a proper async implementation"?

Reply:
It was me being stupid. I'm still not familiar with async/await and did not thought about `write_all().await` while that still could have happened with "old" futures as well. So yeah, me just being stupid.

Reply:
Thanks for clarifying. I thought I may have misunderstood something.

Reply:
Yeah sorry!